### PR TITLE
test material + challenges

### PR DESCRIPTION
**Pull request by one1color**

### CODE REVIEW

File: filter_reduce_sort.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  for dwarf in dwarves where dwarf > "M" {
+    afterM.append(dwarf)
+  }
+  return afterM
+}
+
+
+let dwarvesAfterM2 = arrayOfDwarfArrays.flatMap { dwarves -> [String] in
+  dwarves.filter { $0 > "M" }
+}
```
Comment (TRamson-MRX) [inline filter_reduce_sort.playground/Contents.swift:24]: can you do this without the nesting?

File: filter_reduce_sort.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ Using the same `namesAndAges` dictionary, filter out the adults (those 18 or older) and then use `map` to convert to an array containing just the names (i.e. drop the ages).
+*/
+
+
+
+let adult = namesAndAges.filter {
+    $0.value > 18
+}.map {
+    $0.key
+}
```
Comment (TRamson-MRX) [inline filter_reduce_sort.playground/Contents.swift:130]: Check this challenge again. The answer is not correct.

File: filter_reduce_sort.playground/Contents.swift (hunk 1 above)
Comment (one1color) [reply filter_reduce_sort.playground/Contents.swift:24]: updated

File: filter_reduce_sort.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  return afterM
+}
+
+
+let dwarvesAfterM2 = arrayOfDwarfArrays.flatMap { dwarves -> [String] in
+  dwarves.filter { $0 > "M" }
+}
+
+//: Thomas challenge
+let dwarvesAfterM3 = arrayOfDwarfArrays.flatMap { $0.filter { $0 > "M"} }
```
Comment (TRamson-MRX) [inline filter_reduce_sort.playground/Contents.swift:27]: this is still the same version as M2 just written without the `dwarves -> [String] in`

File: filter_reduce_sort.playground/Contents.swift (hunk 2 above)
Comment (TRamson-MRX) [reply filter_reduce_sort.playground/Contents.swift]: You found it :)

File: filter_reduce_sort.playground/Contents.swift (hunk 3 above)
Comment (one1color) [reply filter_reduce_sort.playground/Contents.swift]: Updated!

**Review by TRamson-MRX — APPROVED** [review]
approve. now it's correct. In 99% of the time you chain such commands and do net nest them into each other. By chaining them it's much easier to add and change things in the logic

**Pull request merged**